timesketch/api/v1/resources/archive.py

```diff
@@ -424,96 +442,364 @@ def _export_view(self, view, sketch, zip_file):
+            sketch: An instance of timesketch.models.sketch.Sketch to unarchive.
+
+        Returns:
+            An integer representing the HTTP status of the operation.
+            - HTTP_STATUS_CODE_OK if successful.
+
+        Raises:
+            HTTPException:
+                - If the sketch is not currently archived.
+                - If any timeline within the sketch is in a state that prevents
+                  unarchiving (e.g., 'processing').
+                - If a critical error occurs while opening an OpenSearch index.
         """
-        if sketch.get_status.status == "archived":
+        logger.info("Unarchiving sketch %s ('%s').", sketch.id, sketch.name)
+
+        # 1. Pre-flight checks
+        if sketch.get_status.status != "archived":
             abort(
                 HTTP_STATUS_CODE_BAD_REQUEST,
-                "Unable to archive a sketch that was already archived "
-                "(sketch: {:d})".format(sketch.id),
+                "Unable to unarchive a sketch that wasn't already archived "
+                f"(sketch: {sketch.id})",
             )
 
-        labels_to_prevent_deletion = current_app.config.get(
-            "LABELS_TO_PREVENT_DELETION", []
-        )
+        errors_occurred = False
+        error_details = []
 
-        for label in labels_to_prevent_deletion:
-            if sketch.has_label(label):
-                abort(
-                    HTTP_STATUS_CODE_FORBIDDEN,
-                    "A sketch with the label {:s} cannot be " "archived.".format(label),
+        # Check if any timeline is in a state that would prevents unarchiving.
+        # TODO enforce after 2026-01-01
+        # https://github.com/google/timesketch/issues/3518
+        statuses_preventing_unarchival = ["processing", "fail"]
+        for timeline in sketch.timelines:
+            timeline_status = timeline.get_status.status
+            if timeline_status in statuses_preventing_unarchival:
+                base_warning_msg = (
+                    f"Unarchiving sketch {sketch.id}, but it contains timeline "
+                    f"'{timeline.name}' (ID: {timeline.id}) in a '{timeline_status}' "
+                    "state. "
                 )
+                specific_advice = ""
+                if timeline_status == "fail":
+                    specific_advice = (
+                        "It is recommended to fix this timeline (e.g., by deleting it) "
+                        "because the sketch cannot be archived again in this state."
+                    )
+                elif timeline_status == "processing":
+                    specific_advice = (
+                        "Please wait for it to finish processing. If it seems to be "
+                        "stuck, contact your system administrator to resolve the issue."
+                    )
+                    # only a processing timeline is stopping the flow
+                    errors_occurred = True
```

Comment:
Why is a processing timeline stopping the unarchive flow? On an old instance there is a chance to run into this situation with an infinite processing timeline being archived. I recommend to handle this similar to what you do with `fail` above.

Reply:
ack